pkg/apis/cr/v1/cluster.go

```diff
@@ -31,9 +31,9 @@ type MySQLClusterSpec struct {
 	// Storage indicates the size of the Persistance Volume Claim for each replica.
 	Storage resource.Quantity `json:"storage"`
 	// Number of mysql instances in the cluster.
-	Replicas uint32 `json:"replicas,omitempty"`
+	Replicas int32 `json:"replicas,omitempty"`
 	// Port specifies port for MySQL server.
-	Port uint16 `json:"port,omitempty"`
+	Port int32 `json:"port,omitempty"`
```

Comment:
I don't see why this would make sense to have negative port or a one greater than `1<<16`